src/VPolygon.jl

```diff
@@ -276,6 +277,123 @@ function ∈(x::AbstractVector{N}, P::VPolygon{N})::Bool where {N<:Real}
     return true
 end
 
+"""
+    rand(::Type{VPolygon}; [N]::Type{<:Real}=Float64, [dim]::Int=2,
+         [rng]::AbstractRNG=GLOBAL_RNG, [seed]::Union{Int, Nothing}=nothing
+        )::VPolygon{N}
+
+Create a random polygon in vertex representation.
+
+### Input
+
+- `VPolygon`     -- type for dispatch
+- `N`            -- (optional, default: `Float64`) numeric type
+- `dim`          -- (optional, default: 2) dimension
+- `rng`          -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`         -- (optional, default: `nothing`) seed for reseeding
+- `num_vertices` -- (optional, default: `-1`) number of vertices of the
+                    polygon (see comment below)
+
+### Output
+
+A random polygon in vertex representation.
+
+### Algorithm
+
+We follow the idea [here](https://stackoverflow.com/a/47358689) based on
+[P. Valtr. Probability that n random points are in convex
+position](https://link.springer.com/article/10.1007%2FBF01271274).
+
+The number of vertices can be controlled with the argument `num_vertices`.
+For a negative value we choose a random number in the range `1:10`.
+"""
+function rand(::Type{VPolygon};
+              N::Type{<:Real}=Float64,
+              dim::Int=2,
+              rng::AbstractRNG=GLOBAL_RNG,
+              seed::Union{Int, Nothing}=nothing,
+              num_vertices::Int=-1
+             )::VPolygon{N}
+    @assert dim == 2 "a polygon must have dimension 2"
+    rng = reseed(rng, seed)
+    if num_vertices < 0
+        num_vertices = rand(1:10)
+    end
+
+    # special cases, 0 or 1 vertex
+    if num_vertices <= 1
+        vertices = [randn(rng, N, 2) for i in 1:num_vertices]
+        return VPolygon(vertices; apply_convex_hull=false)
```

Comment:
the empty constructor (case `num_vertices=0`) does not work with the kwarg `apply_convex_hull`

```julia
julia> VPolygon()
VPolygon{Float64}(Array{Float64,1}[])

julia> VPolygon([]; apply_convex_hull=false)
ERROR: MethodError: no method matching VPolygon(::Array{Any,1}; apply_convex_hull=false)
Closest candidates are:
  VPolygon() at /Users/forets/.julia/dev/LazySets/src/VPolygon.jl:53 got unsupported keyword argument "apply_convex_hull"
  VPolygon(::Array{Array{N<:Real,1},1}; apply_convex_hull, algorithm) where N<:Real at /Users/forets/.julia/dev/LazySets/src/VPolygon.jl:44
Stacktrace:
 [1] top-level scope at none:0
```

don't know if this is a bug; but we can do here

```
if num_vertices == 0
    return VPolygon()    # convex hull is not applied
elseif num_vertices == 1
    return VPolygon([randn(rng, N, 2)])
end
```

Reply:
You have to add a type to the array (as the message suggested):
```julia
julia> VPolygon(Vector{N}[]; apply_convex_hull=false)
VPolygon{Float64}(Array{Float64,1}[])
```